inc/classes/Buffer/class-cache.php

```diff
@@ -669,4 +673,39 @@ private function maybe_dynamic_cookies_filename( $filename, $cookies ) {
 	protected function reset_lowercase( $matches ) {
 		return strtolower( $matches[0] );
 	}
+
+	/**
+	 * Check if permalink structure and url match.
+	 *
+	 * @return bool
+	 */
+	private function maybe_allow_wp_redirect(): bool {
+
+		$permalink_structure = $this->config->get_config( 'permalink_structure' );
+
+		// Last character of permalink.
+		$permalink_last_char = '/' !== substr( $permalink_structure, -1 ) ? '' : '/';
+
+		// Request uri without protocol & domain name.
+		$request_uri = $this->tests->get_request_uri_base();
+
+		// Last character of request uri.
+		$request_uri_last_char = '/' !== substr( $request_uri, -1 ) ? '' : '/';
+
+		$php_self = str_replace( 'index.php', '', $this->config->get_server_input( 'PHP_SELF' ) );
+
+		// In cases where we have the home with a trailng slash (visible or invisible)
+		// and permalink is without trailing slash.
+		if ( '' === $permalink_last_char ) {
+			// Check for root installation.
+			$request_uri_last_char = '/' === $request_uri ? '' : $request_uri_last_char;
+
+			// Check for subdir installation.
+			if ( '/' !== $request_uri && $php_self === $request_uri ) {
+				$request_uri_last_char = '';
+			}
+		}
+
+		return $permalink_last_char === $request_uri_last_char;
```

Comment:
Probably you should return the inverse boolean ( and remove the negation on line81) to be consistent with the function name: currently when _maybe_allow_wp_redirect_ returns true the redirect doesn't happen, while when  it returns false, it redirects

Reply:
@wp-media/php What do you think about this suggestion?